Python/optimizer.c

```diff
@@ -737,6 +737,11 @@ translate_bytecode_to_trace(
                 ADD_TO_TRACE(_TIER2_RESUME_CHECK, 0, 0, target);
                 break;
 
+            case LOAD_GLOBAL:
+                _Py_BloomFilter_Add(dependencies, frame->f_globals);
+                _Py_BloomFilter_Add(dependencies, frame->f_builtins);
+                _Py_FALLTHROUGH;
```

Comment:
I'm not well-versed enough here to be sure at first glance that this is correct. So I'd wait for Mark's input, or anyone else that can confirm this is correct.